wandb/sdk/wandb_config.py

```diff
@@ -153,6 +155,25 @@ def _update(self, d, allow_val_change=None):
         self._items.update(sanitized)
 
     def update(self, d, allow_val_change=None):
+        # check if it's updating existing sweep parameters. if so, throw a specific error.
+        sweep_id = self._settings and self._settings.sweep_id
+        if sweep_id:
+            runs = InternalApi().sweep(sweep_id, "{}")["runs"]
+            sweep_params = dict()
+            for run in runs:
+                if run["name"] == self._settings.run_id:
+                    sweep_config = run["config"]
+                break
+            sweep_params = config_util.parse_config_params(json.loads(sweep_config))
+            for key in d:
+                if key in sweep_params and d[key] != sweep_params[key]:
+                    raise config_util.ConfigError(
+                        (
+                            'You can\'t change the config parameter "{}" from {} to {}'
+                            " because it's part of the sweep.\n"
+                            "If you want to save a new copy, you need to use a new key in config."
+                        ).format(key, sweep_params[key], d[key])
+                    )
```

Comment:
This shouldn't be needed.  We prepopulate the users config when wandb.init is called on an existing run as is the case with sweeps.  We have an existing mechanism `allow_val_change` which defaults to False that will not allow a user to change configuration values that existed when `wandb.init` was called.  We can change the existing error message for allow_val_change to detect if we're in a sweep to provide a better message.  The user can also always opt-in to allowing values to change.  We should inform them they can do this in the error message.

Reply:
allow_val_change is being removed.  _locked attribute should be used on the config object